feat: Add `attachToExisting` features back in for v2

fuzzybinary opened the pull request:
`attachToExisting` is how native apps that add a Flutter component can still interact with Datadog. It was commented out to simplify updating to v2. This change adds the functionality back in..

Also update the hybrid application to work with new interfaces.

refs: RUM-1066

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [x] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [x] Run unit tests
- [x] Run integration tests

fuzzybinary committed:
feat: Add `attachToExisting` features back in for v2

Also update the hybrid application to work with new interfaces.

refs: RUM-1066

0xnm commented on examples/native-hybrid-app/flutter_module/lib/main.dart:
```diff
@@ -2,19 +2,21 @@
-  // )..enableHttpTracking();
+  final config = DatadogAttachConfiguration(
+    detectLongTasks: true,
+    reportFlutterPerformance: true,
```
it is quite a difference with what was before, did we have it enabled before?

fuzzybinary replied:
In the example, no I don't think we did, but probably better to have it enabled.

0xnm commented on packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt:
```diff
@@ -251,17 +253,23 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
-//        )
-//    }
+    private fun attachToExisting(): Map<String, Any> {
+        var loggingEnabled = false
+        if (Logs.isEnabled()) {
+            loggingEnabled = true
+        }
```
it should be possible just to write `val loggingEnabled = Logs.isEnabled()`?

fuzzybinary replied:
Yes. I thought I needed more code here like in RUM.

0xnm commented on packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogSdkPlugin.kt:
```diff
@@ -251,17 +253,23 @@ class DatadogSdkPlugin : FlutterPlugin, MethodCallHandler {
+            loggingEnabled = true
+        }
+
+        var rumEnabled = false
+        if (GlobalRumMonitor.isRegistered()) {
+            rumEnabled = true
```
```suggestion
        val rumEnabled = GlobalRumMonitor.isRegistered()
        if (rumEnabled) {
```

0xnm commented on packages/datadog_flutter_plugin/ios/Classes/DatadogSdkPlugin.swift:
```diff
@@ -294,21 +296,29 @@ public class DatadogSdkPlugin: NSObject, FlutterPlugin {
-    // This is a way to work around https://github.com/flutter/flutter/issues/126671,
-    //
+    private func attachToExisting() -> [String: Any?] {
+        var rumEnabled = false
+        var logsEnabled = false
```
we can probably check if logs and RUM is enabled at the init stage and avoid having `var`

fuzzybinary replied:
Yeah, that's fair. I'll fix.

0xnm commented on packages/datadog_flutter_plugin/lib/src/datadog_configuration.dart:
```diff
@@ -254,104 +254,119 @@ class DatadogConfiguration {
+  /// performance.
+  ///
+  /// Defaults to false
+  bool reportFlutterPerformance = false;
```
should it be a part of this PR? it seems we didn't have it before?

fuzzybinary replied:
We didn't, but that was an oversight. I think adding it here makes sense.

0xnm commented on packages/datadog_flutter_plugin/lib/src/datadog_sdk_method_channel.dart:
```diff
@@ -72,6 +72,8 @@ class DatadogSdkMethodChannel extends DatadogSdkPlatform {
         .invokeMapMethod<String, Object?>(
             'attachToExisting', <String, Object?>{});
 
+    print('channelResponse $channelResponse');
```
some debugging leftover it seems

0xnm commented on packages/datadog_flutter_plugin/lib/src/rum/ddrum.dart:
```diff
@@ -99,39 +102,61 @@ class DatadogRum {
+  }
+
+  DatadogRum._(DatadogSdk core, DatadogRumConfiguration configuration)
+      : tracingSamplingRate = configuration.tracingSamplingRate,
```
I think in native SDKs we try to use `trace` instead of `tracing`, e.g. `traceSampleRate`, so we need to align naming here as well.

0xnm commented on packages/datadog_flutter_plugin/lib/src/rum/ddrum.dart:
```diff
@@ -87,9 +87,12 @@ class DatadogRum {
-  final sampleRandom = Random();
+  /// The sampling rate for tracing resources.
+  ///
+  /// See [RumConfiguration.tracingSamplingRate]
```
I cannot find `RumConfiguration` file in the branch https://github.com/DataDog/dd-sdk-flutter/tree/feature/v2, is it a correct reference? Also it is strange to see notion of `tracing` for RUM here. Should it be in the HTTP plugin support instead only? But I guess it is coming from v1 anyway.

fuzzybinary replied:
Good catch. The configuration was indeed renamed and I missed the doc reference.

Yes, it does only refer to http tracing at the moment, and coming from v1. However, we have several packages that do HTTP tracing and I think it might make sense for them to share the configuration?

fuzzybinary committed:
Fix several suggestions of code review

Fix some documentation (still need to do a full scan) / fix
Rename `tracingSamplingRate` to `traceSampleRate` to be consistent

xgouchet commented on examples/native-hybrid-app/flutter_module/.vscode/launch.json:
```diff
@@ -0,0 +1,13 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
> [!NOTE]
JSON is a data-only format, and using `//` style comments could break some parsing tools (apparently not VSCode 😅 ).
I usually go with an attribute that is ignored and won't break the parsing or code style in editors, e.g.:
```
{
  "comment": "My comment about something"
}
```

fuzzybinary replied:
This is technically `jsonc`, which is literally just json with comments.

fuzzybinary committed:
Perform pod update as part of build

0xnm commented on packages/datadog_flutter_plugin/lib/src/datadog_configuration.dart:
```diff
@@ -254,104 +254,119 @@ class DatadogConfiguration {
+  /// This property only affects network requests made from Flutter, and is not
+  /// shared with or populated from existing the SDK.
+  ///
+  /// See [DdSdkConfiguration.firstPartyHostsWithTracingHeaders] for more information
```
probably a wrong reference?

```suggestion
  /// See [DatadogConfiguration.firstPartyHostsWithTracingHeaders] for more information
```

fuzzybinary replied:
Thanks, I've made task to do a full scan over documentation for renames.

0xnm commented on packages/datadog_flutter_plugin/lib/src/rum/ddrum.dart:
```diff
@@ -87,9 +87,12 @@ class DatadogRum {
   }
 
-  final sampleRandom = Random();
+  /// The sampling rate for tracing resources.
```
```suggestion
  /// The sample rate for tracing resources.
```

0xnm reviewed (commented):
lgtm, just some final suggestions

fuzzybinary committed:
Apply suggestions from code review

Co-authored-by: Nikita Ogorodnikov <4046447+0xnm@users.noreply.github.com>

fuzzybinary committed:
Fix build by adding flutter precache and pod overrides
